### PR TITLE
feat(remap transform): Add `parse_glog` function

### LINKED ISSUE #5372: New `parse_glog` Remap function
Parses [glog](https://medium.com/technical-tips/google-log-glog-output-format-7eb31b3f0ce5)

### PR DESCRIPTION
**Pull request by pablosichert**
Closes #5372.

### CODE REVIEW

File: docs/reference/remap/functions/parse_glog.cue [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package metadata
+
+remap: functions: parse_glog: {
+	category: "Parse"
+	description: """
+		Parses the `value` using the [glog (Google Logging Library)](https://github.com/google/glog) format.
```
Comment (binarylogic) [inline docs/reference/remap/functions/parse_glog.cue:6]:
```suggestion
		Parses the `value` using the [glog (Google Logging Library)](\(urls.glog)) format.
```

Comment (binarylogic) [reply docs/reference/remap/functions/parse_glog.cue:6]: Then we typically add a line to `urls.cue`:

```
glog: "https://github.com/google/glog"
```

This helps us maintain URLs and check that they are still working.

File: tests/behavior/transforms/remap.toml [hunk 2]
```diff
@@ -775,6 +775,32 @@
...
+    type = "log"
+    [tests.input.log_fields]
+      message = "I20210131 14:48:54.411655 15520 main.c++:9] Hello world!"
+  [[tests.outputs]]
+    extract_from = "remap_function_parse_glog"
+    [[tests.outputs.conditions]]
+      type = "remap"
+      source = '''
+        .glog.level == "info" && \
+        .glog.timestamp == to_timestamp!("2021-01-31T14:48:54.411656Z") && \
```
Comment (pablosichert) [inline tests/behavior/transforms/remap.toml:797]: Whoops, left in the changed number at the end to make sure this is run 😄

File: lib/remap-functions/src/parse_glog.rs [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        }
+
+        Ok(log.into())
+    }
+
+    fn type_def(&self, state: &state::Compiler) -> TypeDef {
+        self.value
+            .type_def(state)
+            .fallible_unless(value::Kind::Bytes)
+            .with_constraint(value::Kind::Map)
```
Comment (FungusHumungus) [inline lib/remap-functions/src/parse_glog.rs:116]: Just a heads up, when #6182 lands, this function will need to define an inner type def to specify the fields that the Map will contain. But that PR isn't there yet, so there's nothing you need to do about it now.

**Review by FungusHumungus — APPROVED** [review]
Perfect!

**Pull request merged**